app/src/main/java/org/simple/clinic/summary/SummaryBloodPressurePlaceholderListItem.kt

```diff
@@ -0,0 +1,42 @@
+package org.simple.clinic.summary
+
+import android.view.View
+import android.widget.TextView
+import com.xwray.groupie.Item
+import com.xwray.groupie.ViewHolder
+import io.reactivex.subjects.Subject
+import kotterknife.bindView
+import org.simple.clinic.R
+import org.simple.clinic.widgets.UiEvent
+
+data class SummaryBloodPressurePlaceholderListItem(
+    private val placeholderNumber: Int,
+    private val showHint: Boolean = false
+    // We have three placeholders (might change in the future) so we can't hardcode an id. At the same time, we
+    // we have to choose IDs that will stop it from clashing with other IDs in the adapter, so we multiply the
+    // placeholder number with -20.
+
+    // FIXME: Figure out an automatic solution which will work for all IDs. Also see why we can't use Groupie's
+    // auto generated IDs.
```

Comment:
Groupie auto-generates incrementing IDs which will not be stable. Different items can get the same ID.

Reply:
Coming up with an ID strategy is something I haven't been able to solve properly. What has worked best for me so far is to keep this strategy information in the adapter or a single place so that it's visible to everyone. The controller can then use it for generating Ui models. These IDs usually lie on the negative side to keep them from clashing with actual IDs.

```kotlin
val adapterId = ADAPTER_ID_OFFSET_FOR_BP_PLACEHOLDERS + placeholderIndex
SummaryBloodPressurePlaceholderListItem(adapterId, shouldShowHint)
```